src/c_api.cpp

```diff
@@ -205,8 +207,15 @@ class Booster {
                const Config& config,
                double* out_result, int64_t* out_len) {
     std::lock_guard<std::mutex> lock(mutex_);
-
-    if (single_row_predictor_.get() == nullptr) {
+    if (single_row_predictor_[predict_type].get() == nullptr ||
+	single_row_predictor_pred_early_stop[predict_type] != config.pred_early_stop ||
+	single_row_predictor_pred_early_stop_freq[predict_type] != config.pred_early_stop_freq ||
+	single_row_predictor_pred_early_stop_margin[predict_type] != config.pred_early_stop_margin ||
+	single_row_predictor_iter_[predict_type] != num_iteration) {
+      single_row_predictor_iter_[predict_type] = num_iteration;
+      single_row_predictor_pred_early_stop[predict_type] = config.pred_early_stop;
+      single_row_predictor_pred_early_stop_freq[predict_type] = config.pred_early_stop_freq;
+      single_row_predictor_pred_early_stop_margin[predict_type] = config.pred_early_stop_margin;
```

Comment:
besides of configs, I think the model update (e.g.  more trees) should be considered.

Reply:
I think num_iteration could be always -1 or 0, which means using all models.
So you should better check the real number of trees in the model.